### PR TITLE
Adding Google Authentication to the Application

### PR DESCRIPTION
**Pull request by andy-wojciechowski**
This uses OmniAuth for authentication which then will then authenticate the user using oauth2. Upon successful authentication we create a User object and store it in the session.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class User
+  attr_accessor :id
```
Comment (msoe-sse-webmaster) [inline app/models/user.rb:3]: What is the differentiation between id and name here?

File: config/initializers/omniauth.rb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Rails.application.config.middleware.use OmniAuth::Builder do
+  provider :google_oauth2, ENV['GOOGLE_CLIENT_ID'], ENV['GOOGLE_CLIENT_SECRET'], skip_jwt: true
```
Comment (msoe-sse-webmaster) [inline config/initializers/omniauth.rb:2]: Do we need to put these into Heroku?

Comment (andy-wojciechowski) [reply config/initializers/omniauth.rb:2]: Yes we will need to add them to Heroku for the release. We also will need to make sure the auth callback URL in Heroku works as well when we release

File: app/models/user.rb (hunk 1 above)
Comment (andy-wojciechowski) [reply app/models/user.rb:3]: id represents a uid that gets included in the omniauth request. I think it's the id for the Google user.

name is the name of the Google user. For Example, that property for my account was "Andrew Wojciechowski"

I thought these properties might be useful for when we start limiting logins.

Comment (andy-wojciechowski) [reply app/models/user.rb:3]: I will add some comments explaining what these properties mean in more detail

File: config/initializers/omniauth.rb (hunk 2 above)
Comment (msoe-sse-webmaster) [reply config/initializers/omniauth.rb:2]: Okay, cool. Let me know if you need help with that.

**Review by msoe-sse-webmaster — APPROVED** [review]

**Pull request merged**